tests/src/geometry_checker/testqgsgeometrychecks.cpp

```diff
@@ -1261,9 +1270,23 @@ void TestQgsGeometryChecks::testOverlapCheckToleranceBug()
   // Ensure it actually worked
   testContext.second[layers["overlap_layer_tolerance_bug.shp"]]->getFeature( 0, f );
   QVERIFY( f.geometry().area() < areaOld );
-  // And that we don't have unexpected changes on unaffected points
-  QCOMPARE( f.geometry().vertexAt( 1 ), pointOld_1 );
-  QCOMPARE( f.geometry().vertexAt( 2 ), pointOld_2 );
+  if ( GEOS_VERSION_MAJOR == 3 && GEOS_VERSION_MINOR < 9 )
+  {
+    // And that we don't have unexpected changes on unaffected points
+    QCOMPARE( f.geometry().vertexAt( 1 ), pointOld_1 );
+    QCOMPARE( f.geometry().vertexAt( 2 ), pointOld_2 );
+  }
+  else
+  {
+    /* For reference
+    qDebug() << f.geometry().vertexAt( 1 ).asWkt() << "\n"; // "Point (2537366.84566075634211302 1152360.28978145681321621)"
+    qDebug() << pointOld_1.asWkt() << "\n"; // "Point (2537221.53079314017668366 1152360.02460834058001637)"
+    qDebug() << f.geometry().vertexAt( 2 ).asWkt() << "\n"; // "Point (2537297.08237999258562922 1152290.78251273254863918)"
+    qDebug() << pointOld_2.asWkt() << "\n"; //  "Point (2537366.84566075634211302 1152360.28978145681321621)"
+    */
+    QCOMPARE( f.geometry().vertexAt( 1 ).asWkt( 4 ), QgsGeometry::fromWkt( "Point (2537366.8457 1152360.2898)" ).asWkt( 4 ) );
```

Comment:
Can't these comparisons be done directly against a string literal? Why the need to round trip from/to wkt?

Reply:
Sure. It was to stay in the writing style of the other tests.